modules/effects/arrow-effects-kotlinx-coroutines/src/main/kotlin/arrow/effects/DeferredK.kt

```diff
@@ -502,7 +502,8 @@ sealed class DeferredK<A>(
      */
     fun <A> async(scope: CoroutineScope = GlobalScope, ctx: CoroutineContext = Dispatchers.Default, start: CoroutineStart = CoroutineStart.LAZY, fa: DeferredKProc<A>): DeferredK<A> {
       val conn = DeferredKConnection()
-      val supervisor = Job()
+      //If the context doesn’t have a Job, then the coroutine which is created doesn’t have a parent.
+      val supervisor = scope.coroutineContext[Job] ?: Job()
```

Comment:
`scope.coroutineContext[Job]` what is `Job` here? where is it defined? I'd like to know to understand the architecture better.